eventhubs/client/azure-eventhubs/src/test/java/com/azure/eventhubs/EventDataTest.java

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.eventhubs;
+
+import com.azure.core.amqp.MessageConstant;
+import org.apache.qpid.proton.Proton;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import org.apache.qpid.proton.message.Message;
+import org.junit.Assert;
+import org.junit.Test;
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+
+public class EventDataTest {
+
+    @Test(expected = NullPointerException.class)
+    public void eventDataByteArrayNotNull() {
+        new EventData((byte[]) null);
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void eventDataByteArrayNotNullBuffer() {
+        new EventData((ByteBuffer) null);
+    }
+
+    @Test
+    public void sendingEventsPropsShouldNotBeNull() {
+        final EventData eventData = new EventData("Test".getBytes());
+        Assert.assertNotNull(eventData.systemProperties());
+        Assert.assertNotNull(eventData.body());
+        Assert.assertNotNull(eventData.properties());
+    }
+
+    @Test
+    public void eventDataEmptyByteArray() {
+        final ArrayList<EventData> messages = new ArrayList<>();
+
+        final EventData first = constructMessage(19);
+        final EventData second = constructMessage(22);
+        final EventData third = constructMessage(25);
+        final EventData last = constructMessage(88);
+
+        messages.add(second);
+        messages.add(first);
+        messages.add(last);
+        messages.add(third);
+
+        Collections.sort(messages, (EventData o1, EventData o2) -> o1.sequenceNumber() - o2.sequenceNumber() > 0L ? 1 : -1);
+
+        Assert.assertTrue(messages.get(0) == first);
```

Comment:
`Assert.assertSame(first, messages.get(0));`
And same with the others. It was the comment I had before.